### DISCUSSION TITLE
Call /validator as part of --dry-run

### DISCUSSION THREAD

**Discussion by evanderbent** [Dynatrace/dynatrace-configuration-as-code#243] [Archive] [answered]
Hi,

Currently the validator just checks the internal consistency of the monaco configuration, but it does not check whether the created json object is actually a valid configuration item for Dynatrace.

The use case here is that we would like to work with pull requests and branches, and we would like to run the pipeline and already check whether this configuration would fail when we apply it to Dynatrace (and offload the troubleshooting / debugging to the requestor and avoid non-working configurations inside the main/master branch).

I'm aware that:

- Not every supported/unsupported config API has a validator. As of 1.211 in the configuration api, that is: (/aws/privateLink, /applications/web/{id}/keyUserActions, /applications/web/{id}/errorRules, /plugins/{id}/endpoints/{endpointId}/validator, /extensions/{id}/instances/{configurationId}, /credentials), SLO's in env v2 and synthetic in env v1
- A validator may not actually detect all issues (i.e. it will not eliminate 100% of the incorrect configurations)

Still, I think calling the /validator will save my customers a lot of headache in the future.

**Comment by kristofre** [comment, marked as answer]
I believe this is also described here: https://github.com/dynatrace-oss/dynatrace-monitoring-as-code/discussions/209

**Reply by tatomir146** [reply]
@evanderbent thanks for reporting this, but as @kristofre mentioned this is a duplicate. Please join discussion in #209. Thanks!
